queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/name_request.py

```diff
@@ -62,3 +67,13 @@ def has_new_nr_for_correction(filing: dict):
             get('nameRequest').get('nrNumber', None)
         return old_nr_number != new_nr_number
     return False
+
+
+def has_new_nr_for_alteration(business: Business, filing: dict):
+    """Return whether a alteration filing has new NR."""
+    nr_number = get_str(filing, '/filing/alteration/nameRequest/nrNumber')
+    legal_name = get_str(filing, '/filing/alteration/nameRequest/legalName')
```

Comment:
Just curious, a new NR could potentially have a different legalType as well. 
There's a scenario, where the user starts as a Limited Company, uses a Benefit Company NR and also changes their business type to a Benefit Company. Is that something that should be considered here?

Reply:
thats not relevant here. it's already handled in https://github.com/bcgov/lear/blob/master/queue_services/entity-filer/src/entity_filer/filing_processors/alteration.py#L30

Reply:
Cool!